File: crichere-backend/src/main/kotlin/com/crichere/domain/auth/enums/AuthEnums.kt
package com.crichere.domain.auth.enums

enum class ProfileStatus {
    GHOST, CLAIMED, ACTIVE
}

enum class PlayingRole {
    BATTER, BOWLER, ALL_ROUNDER, WICKET_KEEPER
}

enum class BattingStyle {
    RIGHT_HAND, LEFT_HAND
}

enum class BowlingStyle {
    RIGHT_ARM, LEFT_ARM
}

enum class BowlingType {
    FAST, MEDIUM_FAST, MEDIUM, OFF_SPIN, LEG_SPIN, SLOW_LEFT_ARM, SLOW_LEFT_ARM_ORTHODOX
}

enum class ExperienceLevel {
    LOCAL, DISTRICT, STATE, NATIONAL
}

enum class LeagueRole {
    LEAGUE_ADMIN, AUCTIONEER
}

File: crichere-flutter/lib/features/auth/domain/entities/auth_enums.dart
import 'package:json_annotation/json_annotation.dart';

enum ProfileStatus {
  @JsonValue('GHOST') ghost,
  @JsonValue('CLAIMED') claimed,
  @JsonValue('ACTIVE') active
}

enum PlayingRole {
  @JsonValue('BATTER') batter,
  @JsonValue('BOWLER') bowler,
  @JsonValue('ALL_ROUNDER') allRounder,
  @JsonValue('WICKET_KEEPER') wicketKeeper
}

enum BattingStyle {
  @JsonValue('RIGHT_HAND') rightHand,
  @JsonValue('LEFT_HAND') leftHand
}

enum BowlingStyle {
  @JsonValue('RIGHT_ARM') rightArm,
  @JsonValue('LEFT_ARM') leftArm
}

enum BowlingType {
  @JsonValue('FAST') fast,
  @JsonValue('MEDIUM_FAST') mediumFast,
  @JsonValue('MEDIUM') medium,
  @JsonValue('OFF_SPIN') offSpin,
  @JsonValue('LEG_SPIN') legSpin,
  @JsonValue('SLOW_LEFT_ARM') slowLeftArm,
  @JsonValue('SLOW_LEFT_ARM_ORTHODOX') slowLeftArmOrthodox
}

enum ExperienceLevel {
  @JsonValue('LOCAL') local,
  @JsonValue('DISTRICT') district,
  @JsonValue('STATE') state,
  @JsonValue('NATIONAL') national
}

enum LeagueRole {
  @JsonValue('LEAGUE_ADMIN') leagueAdmin,
  @JsonValue('AUCTIONEER') auctioneer
}
